### PR TITLE
Clarify docs

### PR DESCRIPTION
**Pull request by aochagavia**
I think the previous text was a bit confusing, so decided to reword it in a way that seems clearer to me. I also shortened the comment in the code itself.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,14 +7,19 @@
...
 
-  * `size_of::<SmolStr>() == size_of::<String>()`
-  * Strings up to 22 bytes long do not use heap allocations
-  * Runs of `\n` and space symbols (typical whitespace pattern of indentation
-    in programming laguages) do not use heap allocations
-  * `Clone` is `O(1)`
+* `size_of::<SmolStr>() == size_of::<String>()`
+* Strings of up to 22 bytes are stack-allocated. As an optimization, leading newlines
+are compressed as a single byte, as well as consecutive whitespace found directly after
+the leading newlines. For instance, `"\n\n  hello world"` is stored in 13 bytes,
```
Comment (matklad) [inline README.md]: Ah, unfortunatelly this is not entirely correct: we only compress strings which consist solely of `\n` and ` ` characters. 

To be precise, any substring of [this string](https://github.com/matklad/smol_str/blob/360b36a0e8895e07959d2b65b3a51dbc7c02ab3c/src/lib.rs#L135) will be allocated on the stack.

File: src/lib.rs [hunk 2]
```diff
@@ -1,18 +1,10 @@
...
-///    in programming laguages) do not use heap allocations
-///  * `Clone` is `O(1)`
-///
-/// Unlike `String`, however, `SmolStr` is immutable. The primary use-case for
-/// `SmolStr` is a good enough default storage for tokens of typical programming
-/// languages. A specialized interner might be a better solution for some use-cases.
-///
-/// Intenrally, `SmolStr` is roughly an `enum { Heap<Arc<str>>, Inline([u8; 22]) }`.
+/// * `size_of::<SmolStr>() == size_of::<String>()`
+/// * Strings of up to 22 bytes are stack-allocated (see the readme for details)
```
Comment (matklad) [inline src/lib.rs]: I think it's better to copy-paste the info from readme over here: this is a doc string, so it is available when you browsing the docs on docs.rs or locally (`cargo doc --open`), and Readme is not necessary packaged up with the docs.

File: README.md (hunk 1 above)
Comment (matklad) [reply README.md]: Yes!

**Comment by aochagavia** [comment]
I took the opportunity to clarify the code a bit as well. You seem to be using the `Inline` enum variant in two different ways. Every match statement has a special case for when the `WS_TAG` is set. Given this situation, I think it makes sense to make that special case a proper enum variant, which is what I did (called `Substring`). This does not change the size of `SmolStr`.

I removed the text that says: *Internally, `SmolStr` is roughly an `enum { Heap(Arc<str>), Inline([u8; 22]) }`.* Since there is a third enum variant, I think the best is to omit this for brevity and let people read the code.

**Comment by matklad** [comment]
Oh, excellent simplification! For some reason I’ve thoght I need to manually pack the tag with other bytes.

**Pull request merged**